### PR TITLE
Reorg of ssl.c: CONF, BIO, RAND and EVP_CIPHER

### PR DESCRIPTION
**Pull request by SparkiDev**
Remove whitespace at end of lines in ssl.c.

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -38142,6 +35830,10 @@ int wolfSSL_i2d_ECDSA_SIG(const WOLFSSL_ECDSA_SIG *sig, unsigned char **pp)
 }
 /* End ECDSA_SIG */
 
+
+/* Remove this prototype when code is reorganized. */
+static int wolfSSL_RAND_Init(void);
```
Comment (dgarske) [inline src/ssl.c:35835]: This is already at line 4983. We should only have one prototype.

File: src/ssl.c [hunk 2]
```diff
@@ -54930,315 +51932,3384 @@ int wolfSSL_BUF_MEM_grow(WOLFSSL_BUF_MEM* buf, size_t len)
...
+        return WOLFSSL_NOT_IMPLEMENTED;
+    #endif
+    }
+
+#endif
+
+#if defined(HAVE_LIGHTY) || defined(HAVE_STUNNEL) \
+    || defined(WOLFSSL_MYSQL_COMPATIBLE) || defined(OPENSSL_EXTRA)
+
+WOLFSSL_BIO *wolfSSL_BIO_new_file(const char *filename, const char *mode)
```
Comment (dgarske) [inline src/ssl.c:54309]: This API has different macros guarding it in ssl.h.

```
#if defined(OPENSSL_ALL) \
    || defined(HAVE_STUNNEL) \
    || defined(HAVE_LIGHTY) \
    || defined(WOLFSSL_MYSQL_COMPATIBLE) \
    || defined(WOLFSSL_HAPROXY) \
    || defined(OPENSSL_EXTRA)
```

Have you considered also organizing the header API definitions?

File: src/ssl.c [hunk 3]
```diff
@@ -54930,315 +51932,3384 @@ int wolfSSL_BUF_MEM_grow(WOLFSSL_BUF_MEM* buf, size_t len)
...
+            }
+            data= data->next;
+        }
+    }
+    return (WOLFSSL_STRING*) wolfSSL_lh_retrieve(db->data, value);
+}
+#endif
+
+/*******************************************************************************
+ * END OF TXT_DB API
```
Comment (dgarske) [inline src/ssl.c:52243]: Is it possible to put the new API groups, such that git doesn't think these existing groups have changed?

File: src/ssl.c [hunk 4]
```diff
@@ -54930,315 +51932,3384 @@ int wolfSSL_BUF_MEM_grow(WOLFSSL_BUF_MEM* buf, size_t len)
...
+#ifndef NO_WOLFSSL_STUB
+void wolfSSL_OPENSSL_config(char *config_name)
+{
+    (void)config_name;
+    WOLFSSL_STUB("OPENSSL_config");
+}
+#endif /* !NO_WOLFSSL_STUB */
+#endif /* OPENSSL_ALL || WOLFSSL_NGINX || WOLFSSL_HAPROXY || HAVE_STUNNEL*/
+
+#ifndef NO_CERTS
```
Comment (dgarske) [inline src/ssl.c:52261]: Can we combine these 3 macro into single line and `#if`, since entire group is using same?

**Review by dgarske — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 3 above)
Comment (SparkiDev) [reply src/ssl.c:52243]: I think it was the white space changes that I fixed that stuffed it up.

File: src/ssl.c (hunk 4 above)
Comment (SparkiDev) [reply src/ssl.c:52261]: Done.
We need people to review an update each grouping after reorg.

File: src/ssl.c (hunk 2 above)
Comment (SparkiDev) [reply src/ssl.c:54309]: Fixed wolfSSL_BIO_new_file() and wolfSSL_BIO_new_fp().
We need to assign a person to fix a group bit-by-bit.

File: src/ssl.c (hunk 1 above)
Comment (SparkiDev) [reply src/ssl.c:35835]: That got me too:
4983: static int wolfSSL_RAND_InitMutex(void);
35835: static int wolfSSL_RAND_Init(void);

**Review by dgarske — APPROVED** [review]

**Pull request merged**